### PR TITLE
accounts/usbwallet: use go-hidapi

### PR DESCRIPTION
**Pull request by holiman**
Instead of updating `hidapi` within `karalabe/usb` in https://github.com/karalabe/usb/pull/43, this might be an alternative: use a hidapi-only library from go-ethereum. 

Todo tests: 

- [x] Ledger-devices on Linux (`Nano X`, `Nano S`, `Nano S Plus`) 
- [x] Ledger-devices on Mac 
- [ ] Ledger-devices on windows
- [ ] Trezor on Linux
- [ ] Trezor on Max
- [ ] Trezor on Windows

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,8 @@
 module github.com/ethereum/go-ethereum
 
-go 1.20
+go 1.21
+
+toolchain go1.21.1
```
Comment (holiman) [inline go.mod:5]: My ide started shoving this in there. No idea why.

**Comment by lightclient** [comment]
I verified that Ledger X and Ledger S works on mac.

File: go.mod (hunk 1 above)
Comment (holiman) [inline go.mod:5]:
```suggestion
go 1.20
```

**Comment by holiman** [comment]
Building with this PR requires `libudev`. Fedora `dnf install systemd-devel`, Ubuntu `apt install libhidapi-dev`. Do we add this to the build requirements in the makefile somehow?

**Comment by holiman** [comment]
On my `fedora`, this works well with `Ledger Nano S`, and Ledger Nano X`
This does _not_ work with `Ledger Nano S Plus`. I'm investigating

**Comment by holiman** [comment]
Tagging for triage, 
- Which USB lib to use? 
- How to handle `libudev` build requirement (`systemd-devel`/`libhidapi-dev`)
- Further testing?

**Comment by holiman** [comment]
Closing in favour of https://github.com/ethereum/go-ethereum/pull/28945

**Pull request closed without merging**